api/v1/reviewers/recommendations/ReviewerRecommendationController.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * @file api/v1/reviewers/recommendations/ReviewerRecommendationController.php
```

Comment:
This handler doesn't appear to check that the recommendation being edited/deleted exists in the current context! The manage of one journal can edit and delete recommendations that belong to another.

Reply:
Isn't that handled by `ContextAccessPolicy` defined in `authorize` so that only those have access to context can access that api end point ? And we are restricting the roles to `admin, manager and editor` . 

Or perhaps I am misunderstanding something ?

Reply:
Yes, but you're accepting any recommendation ID as user-supplied data -- and that recommendation might belong to another journal that the manager or editor doesn't manage.